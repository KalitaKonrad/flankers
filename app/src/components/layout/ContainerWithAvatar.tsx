import React, { ComponentProps } from 'react';
import { ImageSourcePropType, StyleSheet, View } from 'react-native';
import { useTheme } from 'react-native-paper';

import { Avatar } from '../shared/Avatar';
import { Container } from './Container';

type ContainerWithAvatarProps = ComponentProps<typeof Container> & {
  avatar: ImageSourcePropType;
  button?: React.ReactNode;
};

const AVATAR_SIZE = 158;

export const ContainerWithAvatar: React.FC<ContainerWithAvatarProps> = (
  props
) => {
  const theme = useTheme();

  return (
    <Container {...props}>
      <View style={styles.avatarContainer}>
        <View
          style={[
            styles.navbarExtension,
            { backgroundColor: theme.colors.primary },
          ]}
        />
        <View style={styles.avatarWrapper}>
          <Avatar
            containerStyle={{ zIndex: 10 }}
            border={4}
            elevation={6}
            size={AVATAR_SIZE}
            src={props.avatar}
          />
          {props.button && <View style={styles.button}>{props.button}</View>}
        </View>
      </View>
      <View style={styles.contentContainer}>{props.children}</View>
    </Container>
  );
};

const styles = StyleSheet.create({
  avatarContainer: {
    alignItems: 'center',
  },
  navbarExtension: {
    width: '100%',
    height: AVATAR_SIZE / 2,
  },
  avatarWrapper: {
    marginTop: -AVATAR_SIZE / 2,
    position: 'relative',
  },
  contentContainer: {
    paddingTop: 24,
  },
  button: {
    elevation: 7,
    position: 'absolute',
    zIndex: 100,
    bottom: 50,
    left: AVATAR_SIZE / 2 + 10,
    backgroundColor: 'red',
  },
});
