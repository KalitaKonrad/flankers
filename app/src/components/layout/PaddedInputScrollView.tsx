import React from 'react';
import { StyleProp, ViewStyle, StyleSheet } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

interface PaddedInputScrollViewProps {
  style?: StyleProp<ViewStyle>;
}

export const PaddedInputScrollView: React.FC<PaddedInputScrollViewProps> = ({
  style,
  children,
}) => {
  return (
    <KeyboardAwareScrollView style={[styles.container, style]}>
      {children}
    </KeyboardAwareScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 8,
    // marginBottom: 110,
    paddingHorizontal: 16,
  },
});
