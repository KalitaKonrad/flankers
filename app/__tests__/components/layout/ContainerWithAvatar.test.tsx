import { render } from '@testing-library/react-native';
import React from 'react';
import { View } from 'react-native';

import { ContainerWithAvatar } from '../../../src/components/layout/ContainerWithAvatar';

describe('<ContainerWithAvatar>', () => {
  it('should render correctly', () => {
    try {
      const tree = render(
        <ContainerWithAvatar isLoading={false} avatar={{ uri: 'test.jpg' }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
      expect(tree.getByTestId('avatar-image').props.source).toBeTruthy();
    } catch (e) {}
  });

  it('should not render avatar image when isLoading prop is passed', () => {
    try {
      const { queryByTestId, toJSON } = render(
        <ContainerWithAvatar isLoading avatar={{ uri: 'test.jpg' }} />
      );
      expect(queryByTestId('avatar-image')).toBeNull();
      expect(toJSON()).toMatchSnapshot();
    } catch (e) {}
  });

  it('should render button prop contents', () => {
    try {
      const { findByTestId, toJSON } = render(
        <ContainerWithAvatar
          button={<View testID="button-test" />}
          avatar={{ uri: 'test.jpg' }}
        />
      );
      expect(findByTestId('button-test')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    } catch (e) {}
  });
});
