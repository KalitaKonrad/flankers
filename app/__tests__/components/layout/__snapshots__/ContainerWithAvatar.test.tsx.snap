// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerWithAvatar> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "position": "relative",
        "zIndex": 100,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 79,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#6200ee",
          },
        ]
      }
    />
    <View
      style={
        Object {
          "marginTop": -79,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderRadius": 100,
              "elevation": 6,
              "height": 158,
              "width": 158,
            },
            Object {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
            },
            Object {
              "zIndex": 10,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://test.example/image.jpg",
            }
          }
          style={
            Array [
              Object {
                "borderColor": "#FFFFFF",
              },
              Object {
                "borderRadius": 100,
                "borderWidth": 4,
                "height": 158,
                "width": 158,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 24,
      }
    }
  />
</View>
`;

exports[`<ContainerWithAvatar> should not render avatar image when isLoading prop is passed 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "position": "relative",
        "zIndex": 100,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 79,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#6200ee",
          },
        ]
      }
    />
    <View
      style={
        Object {
          "marginTop": -79,
          "position": "relative",
        }
      }
    >
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "borderRadius": 100,
                "elevation": 6,
                "height": 158,
                "width": 158,
              },
              Object {
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
              },
              Object {
                "zIndex": 10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <View
              style={
                Array [
                  Array [
                    Object {
                      "borderColor": "#FFFFFF",
                    },
                    Object {
                      "borderRadius": 100,
                      "borderWidth": 4,
                      "height": 158,
                      "width": 158,
                    },
                  ],
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "overflow": "hidden",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 24,
      }
    }
  />
</View>
`;

exports[`<ContainerWithAvatar> should render button prop contents 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "position": "relative",
        "zIndex": 100,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 79,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#6200ee",
          },
        ]
      }
    />
    <View
      style={
        Object {
          "marginTop": -79,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderRadius": 100,
              "elevation": 6,
              "height": 158,
              "width": 158,
            },
            Object {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
            },
            Object {
              "zIndex": 10,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "test.jpg",
            }
          }
          style={
            Array [
              Object {
                "borderColor": "#FFFFFF",
              },
              Object {
                "borderRadius": 100,
                "borderWidth": 4,
                "height": 158,
                "width": 158,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "bottom": 0,
            "elevation": 7,
            "left": 89,
            "position": "absolute",
            "zIndex": 100,
          }
        }
      >
        <View
          testID="button-test"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 24,
      }
    }
  />
</View>
`;
